mailbox/backup/src/test/java/org/apache/james/mailbox/backup/DefaultMailboxBackupTest.java

```diff
@@ -0,0 +1,159 @@
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.HashSet;
+
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import org.apache.commons.compress.utils.SeekableInMemoryByteChannel;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.core.User;
+import org.apache.james.junit.TemporaryFolderExtension;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.backup.ZipAssert.EntryChecks;
+import org.apache.james.mailbox.extension.PreDeletionHook;
+import org.apache.james.mailbox.inmemory.MemoryMailboxManagerProvider;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.MailboxMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import com.github.fge.lambdas.Throwing;
+
+@ExtendWith(TemporaryFolderExtension.class)
+class DefaultMailboxBackupTest implements MailboxMessageFixture {
+
+    private static final String USER = "user";
+    private static final String OTHER_USER = "otherUser";
+
+    private static final User USER1 = User.fromUsername(USER);
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX1 = MailboxPath.forUser(USER, MAILBOX_1_NAME);
+    private static final MailboxPath MAILBOX_PATH_USER1_MAILBOX2 = MailboxPath.forUser(USER, MAILBOX_2_NAME);
+    private static final MailboxPath MAILBOX_PATH_OTHER_USER_MAILBOX1 = MailboxPath.forUser(OTHER_USER, MAILBOX_OTHER_USER_NAME);
+    private static final HashSet<PreDeletionHook> PRE_DELETION_HOOKS = new HashSet<>();
+
+    private final ArchiveService archiveService = new Zipper();
+
+    private MailboxManager mailboxManager;
+    private BlobStore store;
+    private ByteArrayOutputStream destination;
+    private DefaultMailboxBackup backup;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolderExtension.TemporaryFolder temporaryFolder) throws Exception {
+        int bufferSize = 4096;
+        destination = new ByteArrayOutputStream(bufferSize);
+        mailboxManager = MemoryMailboxManagerProvider.provideMailboxManager(PRE_DELETION_HOOKS);
+        store = new MemoryBlobStore(new HashBlobId.Factory());
+        backup = new DefaultMailboxBackup(mailboxManager, archiveService, store);
+    }
+
+    private ZipFile getZipFileFromDestination() throws IOException {
+        byte[] content = destination.toByteArray();
+        SeekableByteChannel byteChannel = new SeekableInMemoryByteChannel(content);
+        return new ZipFile(byteChannel);
+    }
+
+    private void createMailBoxWithMessage(MailboxSession session, MailboxPath mailboxPath, MailboxMessage... messages) throws Exception {
```

Comment:
From the use you do `InputStream...` is the only things needed ^^

No need of that aweful `mailbox/store` data model sneaking in here ;-)

Reply:
I removed the store which effectively wasn't needed anymore.
But i'm not sure to understand you for the mailbox ?

Reply:
Can you change this signature to

```
private void createMailBoxWithMessage(MailboxSession session, MailboxPath mailboxPath, InputStream... messages) 
```

Because it is in test this is not super critical but it will contribute one step further into `using the API, not the implem`.

When it is that easy, we should never hesitate IMO

Reply:
I think i will still use the MailboxMessage. 
This way i will be able to create an AppendCommand including the Flags which i will need other tests.

Reply:
Why not pass the AppendCommand as an argument then?